data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/SqsWorkerIT.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.source.codec.Codec;
+import com.amazon.dataprepper.plugins.source.configuration.OnErrorOption;
+import com.amazon.dataprepper.plugins.source.configuration.SqsOptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SqsWorkerIT {
+    private S3Client s3Client;
+    private SqsClient sqsClient;
+    private S3Service s3Service;
+    private Buffer<Record<Event>> buffer;
+    private S3SourceConfig s3SourceConfig;
+    private S3ObjectGenerator s3ObjectGenerator;
+    private String bucket;
+    private Codec codec;
+    private PluginMetrics pluginMetrics;
+
+    @BeforeEach
+    void setUp() {
+        s3Client = S3Client.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        bucket = System.getProperty("tests.s3source.bucket");
+        s3ObjectGenerator = new S3ObjectGenerator(s3Client, bucket);
+
+
+        sqsClient = SqsClient.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+
+        s3SourceConfig = mock(S3SourceConfig.class);
+        buffer = mock(Buffer.class);
+        codec = mock(Codec.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        s3Service = mock(S3Service.class);
+
+        final SqsOptions sqsOptions = mock(SqsOptions.class);
+        when(sqsOptions.getSqsUrl()).thenReturn(System.getProperty("tests.s3source.queue.url"));
+        when(sqsOptions.getVisibilityTimeout()).thenReturn(Duration.ofSeconds(60));
+        when(sqsOptions.getMaximumMessages()).thenReturn(10);
+        when(sqsOptions.getWaitTime()).thenReturn(Duration.ofSeconds(10));
+        when(s3SourceConfig.getSqsOptions()).thenReturn(sqsOptions);
+        when(s3SourceConfig.getOnErrorOption()).thenReturn(OnErrorOption.DELETE_MESSAGES);
```

Comment:
Any reason we want to mock these objects instead of creating them ? Similarly why do we want to mock S3SourceConfig in line 59 ?

Reply:
My approach here is that it's easier to test config changes if we want to change anything specific to a test, instead of creating a new object  in each test.